blockcypher/src/adapter.ts

```diff
@@ -23,11 +23,15 @@ export const execute: Execute = async (request, config: Config) => {
       const validator = new Validator(request, balance.inputParams)
```

Comment:
We need to split the code for validating the request and parsing input params, from the code to actually execute the fn. Let's just remove the duplicated validation here, and think some more about how to structure these things across adapters.

Reply:
Oh yeah, meant to remove this.

What if each endpoint exposed an execute function and a validate function for custom validation.

Or one further we could go a more Object Oriented route and in `@chainlink/external-adapter` by having an `ExternalAdapter` base class that extends into endpoint specific classes such as `BalanceAdapter`. On this child class would be the custom validation that is shared across all balance adapters (in this case checking `accounts`). This would also let us take out `inputParams` unless it was custom for the specific API.